src/analysis/convex/basic.lean

```diff
@@ -351,6 +351,82 @@ convex_halfspace_gt (is_linear_map.mk complex.add_im complex.smul_im) _
 lemma convex_halfspace_im_lge (r : ℝ) : convex {c : ℂ | r ≤ c.im} :=
 convex_halfspace_ge (is_linear_map.mk complex.add_im complex.smul_im) _
 
+/- Convex combinations in intervals -/
+
+/-
+If x is in an Ioo, it can be expressed as a convex combination of the endpoints.
+-/
+lemma convex_combo_Ioo {a b x : ℝ} :
+    x ∈ Ioo a b → ∃ (x_a x_b : ℝ), 0 < x_a ∧ 0 < x_b ∧ x_a + x_b = 1 ∧ x_a * a + x_b * b = x :=
```

Comment:
I think it would make sense to strengthen all of these to `iff` lemmas, so that they can be used in `rw` and `simp`.

Reply:
Sounds good, I'll do that.